src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt

```diff
@@ -38,8 +36,16 @@ class ImportNameIntention : RsElementBaseIntentionAction<ImportNameIntention.Con
         // TODO: support reexports
         val candidates = RsNamedElementIndex.findElementsByName(project, basePath.referenceName)
             .filterIsInstance<RsQualifiedNamedElement>()
+            .filter { basePath != path || !(it is RsMod || it is RsModDeclItem || it.parent is RsMembers) }
             .mapNotNull {
                 val info = (it as? RsVisible)?.canBeImported(pathSuperMods) ?: return@mapNotNull null
+                // check that result after import can be resolved
+                // if no, don't add it in candidate list
+                if (basePath != path) {
+                    val pkg = info.target.pkg
+                    val fullPath = "${pkg.name}::${info.crateRelativePath.removeSuffix(basePath.text)}${path.text}"
```

Comment:
Yeah, so here's where we start to use `QualifiedNameOwner` in a precise way :) It is ok, but we need to make sure that when we add `trait::function` qualified names (which seems reasonable), we won't try to use them for importing.